File: sample/src/main/java/com/schibsted/spain/barista/sample/SwipeRefreshActivity.java
package com.schibsted.spain.barista.sample;

import android.os.Bundle;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

public class SwipeRefreshActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_swipe_refresh);
    final SwipeRefreshLayout swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.swiperefresh);
    swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
      @Override
      public void onRefresh() {
        TextView textView = (TextView) findViewById(R.id.textview);
        textView.setVisibility(View.VISIBLE);
        swipeRefreshLayout.setRefreshing(false);
      }
    });
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/SwipeRefreshActivity.kt
package com.schibsted.spain.barista.sample

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout

class SwipeRefreshActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_swipe_refresh)
    val swipeRefreshLayout = findViewById<View>(R.id.swiperefresh) as SwipeRefreshLayout
    swipeRefreshLayout.setOnRefreshListener {
      val textView = findViewById<View>(R.id.textview) as TextView
      textView.visibility = View.VISIBLE
      swipeRefreshLayout.isRefreshing = false
    }
  }
}